OneCommunityGlobal/HighestGoodNetworkApp
# Palak - AssignTableRow Unit Test Cases

# Description
Unit test for `AssignTableRow.jsx`

## Main changes explained:
- Added a new file called AssignTableRow.test.js
- Added 8 test cases: 
Renders component with checkbox
Handles checkbox change correctly when initially checked
Handles checkbox change correctly when initially unchecked
Renders without crashing with invalid badge ID in selected badges
Renders without crashing when selectedBadges prop is missing
Renders without crashing with incomplete or invalid badge data
Renders without crashing with an invalid index
Renders without crashing with missing badge image URL

## How to test:
1. check into current branch
2. do `npm test src/__tests__/Badge/AssignTableRow.test.js`
3. check if all the test cases pass

## Screenshots or videos of changes:

![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104417272/0dd6e1a4-f8b6-4c94-b6eb-563e13d204b7)
